File: Xcode/MetricTime/NSDate+MetricTime/NSDate+Minits.m
//
//  NSDate+Secont.m
//  NSDate+Secont
//
//  Created by Joseph Colicchio on 8/10/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

#import "NSDate+Seconts.h"
#import "NSDate+Minits.h"

#define ST_PER_MT 100.0
#define MT_PER_ST (1.0/ST_PER_MT)

@implementation NSDate (Minits)

- (NSMinits)minitsSinceNow {
    return [self secontsSinceNow]*MT_PER_ST;
}
- (NSMinits)minitsSinceReferenceDate {
    return [self secontsSinceReferenceDate]*MT_PER_ST;
}
+ (NSMinits)minitsSinceReferenceDate {
    return [self secontsSinceReferenceDate]*MT_PER_ST;
}
- (NSMinits)minitsSince1970 {
    return [self secontsSince1970]*MT_PER_ST;
}

- (NSMinits)minitsSinceDate:(NSDate *)anotherDate {
    return [self secontsSinceDate:anotherDate]*MT_PER_ST;
}

- (id)addMinits:(NSMinits)minits NS_DEPRECATED(10_0, 10_6, 2_0, 4_0) {
    return [self addSeconts:minits*ST_PER_MT];
}
- (instancetype)dateByAddingMinits:(NSMinits)mt NS_AVAILABLE(10_6, 2_0) {
    return [self dateByAddingSeconts:mt*ST_PER_MT];
}

+ (instancetype)dateWithminitsSinceNow:(NSMinits)mts {
    return [self dateWithSecontsSinceNow:mts*ST_PER_MT];
}
+ (instancetype)dateWithminitsSinceReferenceDate:(NSMinits)mt {
    return [self dateWithSecontsSinceReferenceDate:mt*ST_PER_MT];
}
+ (instancetype)dateWithminitsSince1970:(NSMinits)mts {
    return [self dateWithSecontsSince1970:mts*ST_PER_MT];
}
+ (instancetype)dateWithMinits:(NSMinits)mtsToBeAdded sinceDate:(NSDate *)date {
    return [self dateWithSeconts:mtsToBeAdded*ST_PER_MT sinceDate:date];
}

- (instancetype)initWithminitsSinceNow:(NSMinits)mts {
    self = [self initWithSecontsSinceNow:mts*ST_PER_MT];
    return self;
}
- (instancetype)initWithminitsSinceReferenceDate:(NSMinits)mt {
    self = [self initWithSecontsSinceReferenceDate:mt*ST_PER_MT];
    return self;
}
- (instancetype)initWithminitsSince1970:(NSMinits)mts {
    self = [self initWithSecontsSince1970:mts*ST_PER_MT];
    return self;
}
- (instancetype)initWithMinits:(NSMinits)mtsToBeAdded sinceDate:(NSDate *)date {
    self = [self initWithSeconts:mtsToBeAdded*ST_PER_MT sinceDate:date];
    return self;
}

@end

File: Swift/MetricTime/NSDate+MetricTime/NSDate+Minits.swift
//
//  MetricTimeExtension.swift
//  MetricTime
//
//  Created by Joseph Colicchio on 8/11/16.
//  Copyright © 2016 novacoast. All rights reserved.
//

import Foundation

public typealias Minits = Double;
private let S_PER_MT = (3.0*31556926.0/1000000.0);
private let MT_PER_S = (1.0/S_PER_MT);

public var MinitsSince1970: Minits {
    return NSTimeIntervalSince1970*MT_PER_S;
}

public extension NSDate {
    
    convenience init(minitsSinceReferenceDate:Minits) {
        self.init(timeIntervalSinceReferenceDate:minitsSinceReferenceDate*S_PER_MT);
    }
    convenience init(minitsSinceNow:Minits) {
        self.init(timeIntervalSinceNow:minitsSinceNow*S_PER_MT);
    }
    convenience init(minitsSince1970:Minits) {
        self.init(timeIntervalSince1970:minitsSince1970*S_PER_MT);
    }
    convenience init(minits:Minits, sinceDate:NSDate) {
        self.init(timeInterval:minits*S_PER_MT, sinceDate:sinceDate);
    }
    
    public class func minitsSinceReferenceDate() -> Minits {
        return self.timeIntervalSinceReferenceDate()*MT_PER_S;
    }
    
    func dateByAddingMinits(minits:Minits) -> NSDate {
        return self.dateByAddingTimeInterval(minits*S_PER_MT);
    }
    
    func minitsSinceReferenceDate() -> Minits {
        return self.timeIntervalSinceReferenceDate*MT_PER_S;
    }
    func minitsSinceDate(anotherDate:NSDate) -> Minits {
        return self.timeIntervalSinceDate(anotherDate)*MT_PER_S;
    }
    func minitsSinceNow() -> Minits {
        return self.timeIntervalSinceNow*MT_PER_S;
    }
    func minitsSince1970() -> Minits {
        return self.timeIntervalSince1970*MT_PER_S;
    }
    
}